File: lua/nth-prime/nth-prime.lua
return function(n)
  assert(n > 0, "Error: n must be greater than 0")
  local primes = {2}
  local cur = 3
  while #primes < n do
    local isprime = true
    local max = math.sqrt(cur)
    for _, v in ipairs(primes) do
      if cur % v == 0 then
        isprime = false
        break
      elseif v > max then
        break
      end
    end
    if isprime then
      table.insert(primes, cur)
    end
    cur = cur + 2
  end
  return primes[#primes]
end

File: erlang/nth-prime/src/nth_prime.erl
-module(nth_prime).
-export([prime/1]).

is_prime(N) ->
    is_prime(N, trunc(math:sqrt(N))).
is_prime(_, 1) ->
    1;
is_prime(N, L) when N rem L == 0 ->
    0;
is_prime(N, L) ->
    is_prime(N, L - 1).

prime(0) ->
    error("there is no zeroth prime");
prime(1) ->
    2;
prime(N) ->
    prime(N, 1).
prime(1, P) ->
    P;
prime(N, P) ->
    prime(N - is_prime(P + 2), P + 2).
